### PR TITLE
#2 Update versions for Scala, sbt, plugin and dependencies to latest

### PR DESCRIPTION
**Pull request by ladybunne**
This PR resolves the outdated versions discussed in #2. I'm not entirely confident that this is without problems - sbt is now complaining about the following:

`There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.`

I'm not really sure how to resolve this. Worst-case scenario, this is a decent starting point for the version upgrade.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -2,18 +2,14 @@ organization := "io.github.marad"
...
 resolvers += Resolver.jcenterRepo
 
-seq(bintraySettings:_*)
-seq(bintrayPublishSettings:_*)
-
 libraryDependencies ++= Seq(
-  "org.scalatest" %% "scalatest" % "2.2.1" % "test",
-  "org.scalacheck" %% "scalacheck" % "1.12.2" % "test",
-  "org.mockito" % "mockito-core" % "1.10.19" % "test"
+  "org.scalatest" %% "scalatest" % "3.0.8",
```
Comment (marad) [inline build.sbt]: Why did you remove the `% "test"`? Is this something about the new version?

**Review by marad — COMMENTED** [review]
Hey Kara, thanks for your contribution. I'll take a look at those "evicted" at the weekend.

File: build.sbt (hunk 1 above)
Comment (ladybunne) [reply build.sbt]: I actually mostly did it without thinking (I just pulled the latest version numbers for those libs), but after having researched what that actually does, these seem pointless. I think.

https://www.scala-sbt.org/1.x/docs/Library-Dependencies.html#Per-configuration+dependencies tells more of the story - they're basically dependencies that are configured only to apply on a `test` execution.

You don't have any tests in your project, meaning that you'd never run `test`... meaning that those dependencies are currently doing nothing. Unless you're doing some strange things externally, there's an argument to be made for removing them entirely.

Otherwise I can slap the `test` back on, but may as well remove what you're not using, right?

Comment (marad) [reply build.sbt]: I'm not sure why you think this project has no tests while there are bunch of them here: https://github.com/marad/scalartemis/tree/master/src/test/scala/io/github/scalartemis.

The other thing is... If you remove this `% test` it will add those dependencies to your resulting production JAR making it bigger with no reason, this libraries are only used in tests. So if you really thought there were no tests it would make more sense to remove them entirely.

But there are tests, so please add slap it back on ;)

Comment (ladybunne) [reply build.sbt]: I'm sorry, I had a complete brainfart when I was browsing the codebase - I could've sworn I didn't see a `test` directory... 🤦‍♀ 

I'll add the `% test`s back in, my apologies.

Comment (marad) [reply build.sbt]: No problem :) I haven't got time during the weekend to review the dependency issue yet. I'll try to look at it during the week somehowe.
